index.php

```diff
@@ -25,7 +25,7 @@
 
 \Friendica\Core\Logger\Handler\ErrorHandler::register($dice->create(\Psr\Log\LoggerInterface::class));
 
-$a = \Friendica\DI::app();
+$a = \Friendica\App::fromDice($dice);
```

Comment:
Why can't we use `Dice->create()` here? Like this:
```suggestion
$a = $dice->create(\Friendica\App::class);
```

Reply:
1. Referring to the new Request and Response flow from #14535 we will need the DIC inside the App to get a controller from the DIC. The controller will than be called with the request. The App works as the glue between the HTTP request, the router and the controller that handles the request. We would never need (and will prevent) Dice to create an App instance. If we allow Dice to create an App instance we can not prevent some classes to require it through dependency injection. 
2. Dice is an implementation detail that should not be known inside the app. Instead we will create the App from a PSR-11 Container implementation. This allows us to replace Dice with a different PSR-11 implementation very easily if we like or have to.

Reply:
I understand 1. as a general separation of concern precaution, but we have total control over the code base, so we can enforce the "no requiring App through dependency injection". Additionally, you're on good pace to make App completely useless as a dependency by moving all the global getters/setters to the AppHelper, so it shouldn't happen anyway.

I don't understand 2. though. Is the code you added in this PR PSR-11 compliant? For me it looks like you unpacked a regular call to `$dice->create(\Friendica\App::class);` into its specifics.

Reply:
This PR is WIP right now. I think it would be easier it you look at the atomic commits and see the end result than I could explain with words.